lib/elixir/lib/file.ex

```diff
@@ -931,6 +931,11 @@ defmodule File do
   option when opening the file and then all other functions from
   `IO` are available, since they work directly with Unicode data.
 
+  `modes_or_function` can either be a list of modes or a function. If it's a
+  list, it's considered to be a list of modes (that are documented below). If
+  it's a function, then it's equivalent to calling `open(path, [],
+  modes_or_function)`. See the documentation for `open/3` for more information.
```

Comment:
Isn't it weird that `open/3` says to look at `open/2` for more information as well?

Maybe we are also teaching recursion? 😅  Maybe we should document the function behaviour here as well?